From 43ce5af863a6f653f810ce074f04e2cc1371cc76 Mon Sep 17 00:00:00 2001
From: Jason Rogena <j.rogena@cgiar.org>
Date: Tue, 27 May 2014 20:31:09 +0300
Subject: [PATCH 170/213] Version 0.9.4: Long press to clear date of birth

Users can now long press the date of birth edittext in cowregistration
activity to clear the date
---
 .../cgiar/ilri/mistro/farmer/CowRegistrationActivity.java    | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
index eda2084..02039e6 100644
--- a/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
+++ b/MistroFarmer/src/main/java/org/cgiar/ilri/mistro/farmer/CowRegistrationActivity.java
@@ -55,7 +55,7 @@ import java.util.List;
 
 public class CowRegistrationActivity extends SherlockActivity implements MistroActivity,
         View.OnClickListener, DatePickerDialog.OnDateSetListener, ListView.OnItemClickListener,
-        Spinner.OnItemSelectedListener, View.OnFocusChangeListener, LocationListener, CheckBox.OnCheckedChangeListener
+        Spinner.OnItemSelectedListener, View.OnFocusChangeListener, LocationListener, CheckBox.OnCheckedChangeListener, View.OnLongClickListener
 {
     private boolean cacheData;
 
@@ -154,6 +154,7 @@ public class CowRegistrationActivity extends SherlockActivity implements MistroA
         dateOfBirthET=(EditText)this.findViewById(R.id.date_of_birth_et);
         dateOfBirthET.setOnFocusChangeListener(this);
         dateOfBirthET.setOnClickListener(this);
+        dateOfBirthET.setOnLongClickListener(this);
         //dateOfBirthET.setVisibility(TextView.GONE); //date of birth appears not to be necessary
         breedTV=(TextView)this.findViewById(R.id.breed_tv);
         breedET=(EditText)this.findViewById(R.id.breed_et);
@@ -1218,6 +1219,15 @@ public class CowRegistrationActivity extends SherlockActivity implements MistroA
         }
     }
 
+    @Override
+    public boolean onLongClick(View v) {
+        if(v.equals(dateOfBirthET)){
+            Log.w(TAG, "About to delete date in dateOfBirthET");
+            dateOfBirthET.setText("");
+        }
+        return false;
+    }
+
     private class ServerRegistrationThread extends AsyncTask<JSONObject,Integer,String> {
         private ProgressDialog progressDialog;
 
-- 
2.7.1

